### PR TITLE
[LINKIS #1631] Add linkis-engineplugin-presto module

### PR DESCRIPTION
**Pull request by wForget**
### What is the purpose of the change

PrestoEngine adapts to the Linkis1.X's architecture, Implement the PrestoEngineConn.
#1631

### Brief change log

- Implement the `linkis-engineplugin-presto` module according to the engine development specification

### Verifying this change

- Using `UJESClient`, submit a presto task to verify it.

### Does this pull request potentially affect one of the following parts:
- Dependencies (does it add or upgrade a dependency): (yes)
- Anything that affects deployment: (no)
- The MGS(Microservice Governance Services), i.e., Spring Cloud Gateway, OpenFeign, Eureka.: (no)

### Documentation
- Does this pull request introduce a new feature? (yes)
- If yes, how is the feature documented? (not documented)

### CODE REVIEW

File: linkis-engineconn-plugins/engineconn-plugins/presto/src/main/assembly/distribution.xml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<assembly
+        xmlns="http://maven.apache.org/plugins/maven-assembly-plugin/assembly/2.3"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="http://maven.apache.org/plugins/maven-assembly-plugin/assembly/2.3 http://maven.apache.org/xsd/assembly-1.1.2.xsd">
+    <id>linkis-engineplugin-hive</id>
```
Comment (peacewong) [inline linkis-engineconn-plugins/engineconn-plugins/presto/src/main/assembly/distribution.xml:23]: need rename to presto

File: linkis-engineconn-plugins/engineconn-plugins/presto/src/main/assembly/distribution.xml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        </dependencySet>
+    </dependencySets>
+
+    <fileSets>
+
+        <fileSet>
+            <directory>${basedir}/src/main/resources</directory>
+            <includes>
+                <include>linkis-engineconn.properties</include>
+                <include>log4j2-engineconn.xml</include>
```
Comment (peacewong) [inline linkis-engineconn-plugins/engineconn-plugins/presto/src/main/assembly/distribution.xml:51]: log4j2 need remove -engineconn suffix

Comment (peacewong) [inline linkis-engineconn-plugins/engineconn-plugins/presto/src/main/assembly/distribution.xml:51]: log4j2 need remove -engineconn suffix

File: linkis-engineconn-plugins/engineconn-plugins/presto/src/main/resources/linkis-engineconn.properties [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+wds.linkis.server.version=v1
+
+wds.linkis.engineconn.debug.enable=true
```
Comment (peacewong) [inline linkis-engineconn-plugins/engineconn-plugins/presto/src/main/resources/linkis-engineconn.properties:18]: need remove debug

Comment (peacewong) [inline linkis-engineconn-plugins/engineconn-plugins/presto/src/main/resources/linkis-engineconn.properties:18]: need remove debug

File: linkis-engineconn-plugins/engineconn-plugins/presto/src/main/scala/org/apache/linkis/engineplugin/presto/conf/PrestoConfiguration.scala [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+import org.apache.linkis.common.conf.{ByteType, CommonVars}
+
+object PrestoConfiguration {
+
+  val ENGINE_CONCURRENT_LIMIT = CommonVars[Int]("wds.linkis.engineconn.concurrent.limit", 100)
+
+  val ENTRANCE_MAX_JOB_INSTANCE = CommonVars[Int]("wds.linkis.entrance.max.job.instance", 100)
+  val ENTRANCE_PROTECTED_JOB_INSTANCE = CommonVars[Int]("wds.linkis.entrance.protected.job.instance", 0)
+  val ENTRANCE_RESULTS_MAX_CACHE = CommonVars("wds.linkis.presto.resultSet.cache.max", new ByteType("512k"))
```
Comment (peacewong) [inline linkis-engineconn-plugins/engineconn-plugins/presto/src/main/scala/org/apache/linkis/engineplugin/presto/conf/PrestoConfiguration.scala:29]: need add var desc

Comment (peacewong) [inline linkis-engineconn-plugins/engineconn-plugins/presto/src/main/scala/org/apache/linkis/engineplugin/presto/conf/PrestoConfiguration.scala:29]: need add var desc

File: linkis-engineconn-plugins/engineconn-plugins/presto/src/main/scala/org/apache/linkis/engineplugin/presto/factory/PrestoEngineConnResourceFactory.scala [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+import java.util
+
+import org.apache.linkis.common.utils.Logging
+import org.apache.linkis.manager.common.entity.resource.Resource
+import org.apache.linkis.manager.engineplugin.common.resource.AbstractEngineResourceFactory
+
+class PrestoEngineConnResourceFactory extends AbstractEngineResourceFactory with Logging {
+
+  override protected def getRequestResource(properties: util.Map[String, String]): Resource = {
```
Comment (peacewong) [inline linkis-engineconn-plugins/engineconn-plugins/presto/src/main/scala/org/apache/linkis/engineplugin/presto/factory/PrestoEngineConnResourceFactory.scala:27]: need remove this file

Comment (peacewong) [inline linkis-engineconn-plugins/engineconn-plugins/presto/src/main/scala/org/apache/linkis/engineplugin/presto/factory/PrestoEngineConnResourceFactory.scala:27]: need remove this file

File: linkis-engineconn-plugins/engineconn-plugins/presto/src/main/assembly/distribution.xml (hunk 2 above)
Comment (wForget) [reply linkis-engineconn-plugins/engineconn-plugins/presto/src/main/assembly/distribution.xml]: fixed

File: linkis-engineconn-plugins/engineconn-plugins/presto/src/main/resources/linkis-engineconn.properties (hunk 3 above)
Comment (wForget) [reply linkis-engineconn-plugins/engineconn-plugins/presto/src/main/resources/linkis-engineconn.properties]: fixed

File: linkis-engineconn-plugins/engineconn-plugins/presto/src/main/scala/org/apache/linkis/engineplugin/presto/factory/PrestoEngineConnResourceFactory.scala (hunk 5 above)
Comment (wForget) [reply linkis-engineconn-plugins/engineconn-plugins/presto/src/main/scala/org/apache/linkis/engineplugin/presto/factory/PrestoEngineConnResourceFactory.scala]: fixed

**Review by peacewong — APPROVED** [review]
LGTM.

**Pull request merged**